modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/resource/test/HeadlessBuilderResourceTest.java

```diff
@@ -3123,6 +3067,194 @@ private void _relateObjectEntries(
 			objectRelationship, TestPropsValues.getUserId());
 	}
 
+	private void _testGetInDifferentCompany(
+			String apiSchemaExternalReferenceCode,
+			ObjectDefinition objectDefinition, long userId)
+		throws Exception {
+
+		HTTPTestUtil.customize(
+		).withBaseURL(
+			"http://www.able.com:8080"
+		).withCredentials(
+			"test@able.com", "test"
+		).apply(
+			() -> {
+				assertSuccessfulJSONObject(
+					JSONUtil.put(
+						"apiApplicationToAPIEndpoints",
+						JSONUtil.putAll(
+							_createAPIEndpoint(
+								_API_ENDPOINT_ERC_1, Http.Method.GET,
+								_API_APPLICATION_PATH_1, null,
+								APIApplication.Endpoint.RetrieveType.COLLECTION.
+									getValue(),
+								APIApplication.Endpoint.Scope.COMPANY))
+					).put(
+						"apiApplicationToAPISchemas",
+						JSONUtil.put(
+							JSONUtil.put(
+								"apiSchemaToAPIProperties",
+								JSONUtil.putAll(
+									JSONUtil.put(
+										"description", "description"
+									).put(
+										"name", "textProperty"
+									).put(
+										"objectFieldERC",
+										_API_SCHEMA_TEXT_FIELD_ERC
+									))
+							).put(
+								"description", "description"
+							).put(
+								"externalReferenceCode",
+								apiSchemaExternalReferenceCode
+							).put(
+								"mainObjectDefinitionERC",
+								objectDefinition.getExternalReferenceCode()
+							).put(
+								"name", "name"
+							))
+					).put(
+						"applicationStatus", "published"
+					).put(
+						"baseURL", _BASE_URL_1
+					).put(
+						"externalReferenceCode", _API_APPLICATION_ERC_1
+					).put(
+						"title", RandomTestUtil.randomString()
+					).toString(),
+					"headless-builder/applications", Http.Method.POST);
+				assertSuccessfulJSONObject(
+					null,
+					StringBundler.concat(
+						"headless-builder/schemas/by-external-reference-code/",
+						apiSchemaExternalReferenceCode,
+						"/responseAPISchemaToAPIEndpoints/",
+						_API_ENDPOINT_ERC_1),
+					Http.Method.PUT);
+
+				String textFieldValue = RandomTestUtil.randomString();
+
+				ObjectEntryTestUtil.addObjectEntry(
+					objectDefinition, userId,
+					HashMapBuilder.<String, Serializable>put(
+						"textField", textFieldValue
+					).build());
+
+				String jsonObjectString = HTTPTestUtil.invokeToJSONObject(
+					null, "c/" + _BASE_URL_1 + _API_APPLICATION_PATH_1,
+					Http.Method.GET
+				).toString();
+
+				JSONAssert.assertEquals(
+					JSONUtil.put(
+						"items",
+						JSONUtil.putAll(
+							JSONUtil.put("textProperty", textFieldValue))
+					).put(
+						"lastPage", 1
+					).put(
+						"page", 1
+					).put(
+						"pageSize", 20
+					).put(
+						"totalCount", 1
+					).toString(),
+					jsonObjectString, JSONCompareMode.LENIENT);
+			}
+		);
+	}
+
+	private void _testGetOpenAPIInDifferentCompany() throws Exception {
```

Comment:
Can we simplify this test and use a testGet method in order to validate in the given company passed by argument? I think the result should be the same no matter which company is